src/main/java/com/aws/greengrass/device/iot/CertificateRegistry.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.device.iot;
+
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.util.Digest;
+import com.aws.greengrass.util.Utils;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+import javax.inject.Inject;
+
+import static com.aws.greengrass.device.ClientDevicesAuthService.DEFAULT_MAX_ACTIVE_AUTH_TOKENS;
+
+
+public class CertificateRegistry {
+    private static final Logger logger = LogManager.getLogger(CertificateRegistry.class);
+    public static final int REGISTRY_CACHE_SIZE = DEFAULT_MAX_ACTIVE_AUTH_TOKENS;
+    // holds mapping of certificateHash (SHA-256 hash of certificatePem) to IoT Certificate Id;
+    // size-bound by default cache size, evicts oldest written entry if the max size is reached
+    private static final Map<String, String> certificateHashToIdMap = Collections.synchronizedMap(
+            new LinkedHashMap<>(REGISTRY_CACHE_SIZE, 0.75f, false));
```

Comment:
this creates a map with an initial capacity, but not a fixed capacity, right? How are we stopping this from growing forever?

Reply:
https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html#removeEldestEntry-java.util.Map.Entry-

Default is `false`, so it will grow forever.

Reply:
With a specific max size, we can just use a load factor of 1. It can allocate that size and never need to change it.

Reply:
right, missed to add the size check for eviction. Updated now. Thanks.